### DISCUSSION TITLE
[@types/validator] NodeNext requires accessing everything via validator.default

### DISCUSSION THREAD

**Discussion by benasher44** [DefinitelyTyped/DefinitelyTyped#65343] [Issues with a @types package]
With moduleResolution set to NodeNext in tsconfig.json, you can only access validator functions via validator.default. For example:

```typescript
import validator from "validator";

validator.default.isURL(…
```

Would it be possible to update types to better support NodeNext?

[a comment by an automated account is not shown]

**Comment by benasher44** [comment]
@andrewbranch I think this one can just be updated from `export default validator` to `export = validator`, is that right?

**Reply by andrewbranch** [reply]
The JS file has

```ts
var _default = validator;
exports.default = _default;
module.exports = exports.default;
module.exports.default = exports.default;
```

So ideally the types need to `export =` something that also has a property `default` which has the same type as the `export =`. This can be done by declaring a `default` property on whatever object gets exported (e.g. a `static` one if it’s a class) or by merging with a namespace that exports a constant by the name `default`. Haven’t looked at the existing validator types to know exactly what that refactor would look like. Happy to help if needed.

**Comment by benasher44** [comment]
Fixed in #68121. Thanks for your help @andrewbranch!
